src/cli/mod.rs

```diff
@@ -0,0 +1,241 @@
+mod paths;
+mod prove_and_verify;
+mod repl;
+
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+
+use lurk::eval::lang::Coproc;
+use lurk::field::{LanguageField, LurkField};
+use lurk::store::Store;
+use lurk::z_data::{from_z_data, ZData};
+use lurk::z_store::ZStore;
+use pasta_curves::{pallas, vesta};
+
+use clap::{Args, Parser, Subcommand};
+use tap::TapOptional;
+
+use self::prove_and_verify::verify_proof;
+use self::repl::Repl;
+
+const DEFAULT_FIELD: LanguageField = LanguageField::Pallas;
+const DEFAULT_LIMIT: usize = 100_000_000;
+const DEFAULT_RC: usize = 1;
+
+#[derive(Parser, Debug)]
+#[clap(version)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+enum Command {
+    /// Loads a file, processing forms sequentially ("load" can be elided)
+    Load(LoadArgs),
+    /// Enters Lurk's REPL environment ("repl" can be elided)
+    Repl(ReplArgs),
+    /// Verifies a Lurk proof
+    Verify(VerifyArgs),
+}
+
+#[derive(Args, Debug)]
+struct LoadArgs {
+    /// The file to be loaded
+    #[clap(value_parser)]
+    lurk_file: PathBuf,
+
+    /// Path to a custom ZStore to be used
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    /// Maximum number of iterations allowed (defaults to 100_000_000)
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+
+    /// Reduction count used for proofs (defaults to 1)
+    #[clap(long, value_parser)]
+    rc: Option<usize>,
+
+    /// Flag to prove the last evaluation
+    #[arg(long)]
+    prove: bool,
+}
+
+#[derive(Parser, Debug)]
+struct LoadCli {
+    #[clap(value_parser)]
+    lurk_file: PathBuf,
+
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+
+    #[arg(long)]
+    prove: bool,
+
+    #[clap(long, value_parser)]
+    rc: Option<usize>,
+}
+
+impl LoadArgs {
+    pub fn into_cli(self) -> LoadCli {
+        LoadCli {
+            lurk_file: self.lurk_file,
+            zstore: self.zstore,
+            limit: self.limit,
+            prove: self.prove,
+            rc: self.rc,
+        }
+    }
+}
+
+#[derive(Args, Debug)]
+struct ReplArgs {
+    /// Path to a custom ZStore to be used
```

Comment:
What does it mean to 'use' a `ZStore`? Will the data at this path be modified in place as the in-memory store evolves? Will it be updated at some point? Or does this just mean that the in-memory store will initially contain the data in the `ZStore`? Documenting this would be useful.

Reply:
I've improved the description